pilot/cmd/pilot-agent/main.go

```diff
@@ -568,6 +572,16 @@ var (
 	}
 )
 
+func getProxyConfig() (meshconfig.ProxyConfig, error) {
+	defaultConfig := mesh.DefaultProxyConfig()
+	if proxyConfigEnv != "" {
+		if err := gogoprotomarshal.ApplyJSON(proxyConfigEnv, &defaultConfig); err != nil {
```

Comment:
Couple Questions - I am not fully familiar with `ApplyJSON`
- Would `ApplyJSON ` merge values to defaultConfig? So if `proxyConfigEnv` does not specify a value would this get merged with the default value? Was that the intention of passing defaultConfig here?
- Should we return `nil` incase of error - because at the call site we do not seem to use the returned defaultconfig ?

Reply:
we don't have  pointer so we can't return nil, although I supposed as could return a empty struct. not sure which is idiomatic.

Yes apply will do merging, so defaults are preserved. Note this is how pilot behaves as well

Reply:
> we don't have pointer so we can't return nil, although I supposed as could return a empty struct. not sure which is idiomatic.

I see. Not a big deal here. We can leave it as is

Reply:
One option is to return a pointer from this func - not sure if that is more idiomatic, but up to you NBD